src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +163,21 @@ class bo : public xrt::bo
   }
 };
 
+class event_impl;
+class event 
+{
+public:
+
+  event(const xrt::device& device, int option, const std::string& port1_name, const std::string& port2_name, uint32_t value);
```

Comment:
instead of constructor taking all the variables, Can you have constructor which takes only device class 
and have another member function to start the profiling.